Makefile

```diff
@@ -410,60 +427,55 @@ tools/$(HTML_COMPRESSOR):
 else
 tools/$(HTML_COMPRESSOR):
 	$(Q) mkdir -p tools
-	cd tools; wget https://github.com/yui/yuicompressor/releases/download/v2.4.8/$(YUI_COMPRESSOR)
-	cd tools; wget https://htmlcompressor.googlecode.com/files/$(HTML_COMPRESSOR)
+	$(Q) if ! test -f tools/$(YUI_COMPRESSOR); then wget https://github.com/yui/yuicompressor/releases/download/v2.4.8/$(YUI_COMPRESSOR) -O tools/$(YUI_COMPRESSOR); fi
+	$(Q) if ! test -f tools/$(HTML_COMPRESSOR); then wget https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/htmlcompressor/$(HTML_COMPRESSOR) -O tools/$(HTML_COMPRESSOR); fi 
 endif
 
 ifeq ("$(COMPRESS_W_HTMLCOMPRESSOR)","yes")
 $(BUILD_BASE)/espfs_img.o: tools/$(HTML_COMPRESSOR)
 endif
 
-$(BUILD_BASE)/espfs_img.o: html/ html/wifi/ espfs/mkespfsimage/mkespfsimage
-	$(Q) rm -rf html_compressed; mkdir html_compressed; mkdir html_compressed/wifi;
-	$(Q) cp -r html/*.ico html_compressed;
-	$(Q) cp -r html/*.css html_compressed;
-	$(Q) cp -r html/*.js html_compressed;
-	$(Q) cp -r html/wifi/*.png html_compressed/wifi;
-	$(Q) cp -r html/wifi/*.js html_compressed/wifi;
+$(BUILD_BASE)/espfs_img.o: $(shell find html) espfs/mkespfsimage/mkespfsimage
+	$(Q) rm -rf html_compressed; mkdir -p html_compressed; cp -r html/* html_compressed; 
 ifeq ("$(COMPRESS_W_HTMLCOMPRESSOR)","yes")
-	$(Q) echo "Compression assets with htmlcompressor. This may take a while..."
-		$(Q) java -jar tools/$(HTML_COMPRESSOR) \
-		-t html --remove-surrounding-spaces max --remove-quotes --remove-intertag-spaces \
-		-o $(abspath ./html_compressed)/ \
-		$(HTML_PATH)head- \
-		$(HTML_PATH)*.html
-	$(Q) java -jar tools/$(HTML_COMPRESSOR) \
-		-t html --remove-surrounding-spaces max --remove-quotes --remove-intertag-spaces \
-		-o $(abspath ./html_compressed)/wifi/ \
-		$(WIFI_PATH)*.html
-	$(Q) echo "Compression assets with yui-compressor. This may take a while..."
+	$(Q) echo "Compression assets with htmlcompressor. This is fast..."
+	$(Q) for file in `find html_compressed -type f -name "*.html"`; do \
+			java -jar yui/$(HTML_COMPRESSOR) \
+			-t html --remove-surrounding-spaces max --remove-quotes --remove-intertag-spaces \
+			-o $$file $$file; \
+		done
+	$(Q) echo "Compressing JS/CSS with yui-compressor. This may take a while..."
```

Comment:
Adding SSL library is optional - I had to send data to a remote server using HTTPS.
Disabling warnings for unused variables is also optional.
Adding compiler defines for date, time and build number is nedded in order to make these values available to the C source code.
Creating an auto-incrementing build number actually involves creating a new BUILD_NUMBER_SRC file every time the build number changes - so that the C file has the newest timestamp and this forces the **make** to rebuild its "older" object file.
There is a make rule which tells that BUILD_NUMBER_FILE depends on USER1_OUT - so for every change of the firmware the build number auto-increments.
Instead of echoing instructions to user how to dump and dis-assemble the firmware, I put an actual command that dumps dis-assembled info to a file with extension ".dump". This is easier for the user.
The make rule for "user1.bin" states that it depends also on BUILD_NUMBER_FILE.
"ls -ls eagle*bin" is of doubtful use.
Showing how many bytes from the flash uses **user1.bin** will be even more useful if you show this number as a percentage of the whole available size - it is much easier for a human to comprehend 2-digit percentage than 6 digits number in bytes.
Downloading of YUI and HTML compressor utilities is only needed if they are not yet downloaded - there is no need to download them **every** time I run the **make** command.
Instead of manually listing which HTML, JS, CSS files to compress - it is more logical and future-proof to enumerate them automatically. When you add new files - there will be no need to edit the Makefile, the **find** command will take care of this.
I just saw a mistake - the **yui** folder should be actually named **tools**. Sorry, this is copy-pasted from my Makefile.